conans/client/installer.py

```diff
@@ -316,16 +317,38 @@ def _node_concurrently_installed(self, node, package_folder):
             if node.update_manifest == read_manifest:
                 return True
 
+    def _load_editables_cpp_info(self):
+        editables_path = self._client_cache.default_editable_path
+        if os.path.exists(editables_path):
+            return EditableCppInfo.create(editables_path, require_namespace=True)
+        return None
+
     def _handle_node_editable(self, node):
+        # Get source of information
         package_layout = self._client_cache.package_layout(node.conan_ref)
-        package_layout_file = package_layout.editable_package_layout_file()
-        base_path = os.path.dirname(package_layout_file)
-
+        base_path = package_layout.conan()
         self._call_package_info(node.conanfile, package_folder=base_path)
-        editable_cpp_info = EditableCppInfo.create(package_layout_file, base_path=base_path,
-                                                   settings=node.conanfile.settings,
-                                                   options=node.conanfile.options)
-        editable_cpp_info.apply_to(node.conanfile.cpp_info)
+
+        if self._editable_cpp_info and self._editable_cpp_info.has_info_for(node.conanfile.name):
+            # Try with the profile-like file
+            self._editable_cpp_info.apply_to(node.conanfile.name,
+                                             node.conanfile.cpp_info,
+                                             base_path=base_path,
+                                             settings=node.conanfile.settings,
+                                             options=node.conanfile.options)
+        else:
+            # Try with package-provided file
+            package_layout_file = package_layout.editable_package_layout_file()
+            if os.path.exists(package_layout_file):
+                editable_cpp_info = EditableCppInfo.create(package_layout_file,
+                                                           require_namespace=False)
+                editable_cpp_info.apply_to(node.conanfile.name,
```

Comment:
You might want to factor out the call to ``apply_to()`` and do a single call (restrict the if-else to obtain a ``editable_cpp_info``, then apply it.